test/spec.index.js

```diff
@@ -927,6 +940,154 @@ describe('Template Mixins', function () {
+                res.locals['checkbox-group']().call(res.locals, 'field-name');
+                render.should.have.been.calledWith(sinon.match({
+                    className: 'abc def'
+                }));
+            });
+
+            it('should have role: group', function () {
+                middleware = mixins({
+                    'field-name': {
+                    }
+                });
+                middleware(req, res, next);
+                res.locals['checkbox-group']().call(res.locals, 'field-name');
+                render.should.have.been.calledWith(sinon.match({
+                    role: 'group'
+                }));
+            });
+
+            it('adds `legendClassName` if it exists as a string or an array', function () {
+                middleware = mixins({
+                    'field-name-1': {
+                        legend: {
+                            className: 'abc def'
+                        }
+                    },
+                    'field-name-2': {
+                        legend: {
+                            className: ['abc', 'def']
+                        }
+                    }
+                });
+
+                middleware(req, res, next);
+
+                res.locals['checkbox-group']().call(res.locals, 'field-name-1');
+                render.should.have.been.calledWith(sinon.match({
+                    legendClassName: 'abc def'
+                }));
+
+                res.locals['checkbox-group']().call(res.locals, 'field-name-2');
+                render.should.have.been.calledWith(sinon.match({
+                    legendClassName: 'abc def'
+                }));
+            });
+
+            it('uses locales translation for legend if a field value isn\'t provided', function () {
+                var translate = sinon.stub().withArgs('fields.field-name.legend').returns('Field legend');
+                middleware = mixins({
+                    'field-name': {}
+                }, { translate: translate });
+                middleware(req, res, next);
+                res.locals['checkbox-group']().call(res.locals, 'field-name');
+                render.should.have.been.calledWithExactly(sinon.match({
+                    legend: 'Field legend'
+                }));
+            });
+
+            it('uses locales translation for hint if a field value isn\'t provided', function () {
+                var translate = sinon.stub().withArgs('fields.field-name.hint').returns('Field hint');
+                middleware = mixins({
+                    'field-name': {}
+                }, { translate: translate });
+                middleware(req, res, next);
+                res.locals['checkbox-group']().call(res.locals, 'field-name');
+                render.should.have.been.calledWithExactly(sinon.match({
+                    hint: 'Field hint'
+                }));
+            });
+
+            it('doesn\'t add a hint if the hint doesn\'t exist in locales', function () {
```

Comment:
you can enter a custom key in field.hint - if you enter custom copy it will try to look it up, fail and print out the copy you entered